Rest.Services.Test/ProductRepositoryTest.cs

```diff
@@ -0,0 +1,70 @@
+using Xunit;
+using Rest.Models;
+using Rest.Services.Test.Utils.Fixtures;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rest.Services.Test;
+
+public class ProductRepositoryTest
+{
+    [Fact]
+    public void Add_OneProduct_ShouldCountOneProduct()
+    {
+        var productRepository = new ProductRepository();
+        var product = new ProductFixtures().BuildProduct();
+        productRepository.Add(product);
+
+        Assert.Equal(1, productRepository.Count());
+    }
+
+    [Fact]
+    public void Remove_OneProduct_ShouldCountOneProducts()
+    {
+        var productRepository = new ProductRepository();
+        var products = new ProductFixtures().BuildProducts(2);
+        productRepository.Add(products[0]);
+        productRepository.Add(products[1]);
+
+        productRepository.Remove(1);
+
+        Assert.Equal(1, productRepository.Count());
+    }
+
+    [Fact]
+    public void Get_OneProduct_ShouldReturnOneProduct()
+    {
+        var productRepository = new ProductRepository();
+        var product = new ProductFixtures().BuildProduct();
+        productRepository.Add(product);
+
+        var result = productRepository.Get(1);
+
+        Assert.Equal(product, result);
+    }
+
+    [Fact]
+    public void Get_NoProduct_ShouldReturnKeyNotFoundException()
+    {
+        var productRepository = new ProductRepository();
+
+        Assert.Throws<KeyNotFoundException>(() => productRepository.Get(2));
+    }
+
+    [Fact]
+    public void GetAll_Products_ShouldReturnThreeProducts()
+    {
+        var productRepository = new ProductRepository();
+        var products = new ProductFixtures().BuildProducts(3);
+
+        for (int i = 0; i < products.Count(); i++)

```

Comment:
Why don't you literally call add 3 times? I don't think for loop is needed at all, a least for a test case of 3 products

Reply:
Get it. I'm just going to call the add method three times.